streaming-connect-common/src/main/java/com/adobe/platform/streaming/http/HttpConnection.java

```diff
@@ -115,22 +115,22 @@ HttpURLConnection connect() throws HttpException {
           close();
           HttpUtil.sleepUninterrupted(retryBackoff);
         } else {
-          throw new HttpException("request failed (" + responseCode + "): " + errorMsg);
+          throw new HttpException("request failed (" + responseCode + "): " + errorMsg, responseCode);
         }
       } catch (MalformedURLException e) {
-        throw new HttpException(("bad withUrl: " + url), e);
+        throw new HttpException(("bad withUrl: " + url), 400, e);
       } catch (IOException e) {
         LOG.warn("attempt {} of {} failed with exception - {}", retries, maxRetries, e.getMessage());
         close();
         cause = e;
         HttpUtil.sleepUninterrupted(retryBackoff);
       } catch (AuthException authException) {
-        throw new HttpException("exception while fetching the auth token", authException);
+        throw new HttpException("exception while fetching the auth token", 401, authException);
       }
     }
 
     if (conn == null) {
-      throw new HttpException("unable to connect", cause);
+      throw new HttpException("unable to connect", 500, cause);
```

Comment:
Can we set default error code as 500 in HttpException ?

Reply:
Sure, good idea